### PR TITLE
Improve docs for Alpha

### PR DESCRIPTION
**Pull request by preston-evans98**
# Description
Improve docs for Alpha

### CODE REVIEW

**Review by bkolad — APPROVED** [review]

File: examples/demo-nft-module/README.md [hunk 1]
```diff
@@ -174,9 +162,15 @@ Before we start implementing the `Module` trait, there are several preparatory s
 
     ```
 
-3.  Define Config. In this case, config will contain admin and initial tokens:
+    As you can see, we derive the `borsh` serialization format for these messages. Unlike most serialization libraries,
+    `borsh` guarantees that all messages have a single "canonical" serialization, which makes it easy to reliably
+    hash and compare serialized messages.
+
+3.  Create a Config struct for the genesis configuration. In this case, the admin address and initial token distribution
```
Comment (cemozerr) [inline examples/demo-nft-module/README.md:169]:
```suggestion
3.  Create a `Config` struct for the genesis configuration. In this case, the admin address and initial token distribution
```

File: examples/demo-nft-module/README.md [hunk 2]
```diff
@@ -147,11 +134,12 @@ Before we start implementing the `Module` trait, there are several preparatory s
     This step is necessary to optimize the module for execution in ZK mode, where none of the RPC-related logic is
     needed.
     Zero Knowledge mode uses a different serialization format, so serde is not needed.
-    The `sov-state` module maintains same logic, so its `native` flag only enabled in that case.
+    The `sov-state` module maintains the same logic, so its `native` flag is only enabled in that case.
 
 2.  Define `Call` messages, which are used to change the state of the module.
```
Comment (cemozerr) [inline examples/demo-nft-module/README.md:139]: Small nit, to be congruent with other steps.

```suggestion
2.  Define `Call` messages, which are used to change the state of the module:
```

File: examples/demo-nft-module/README.md [hunk 3]
```diff
@@ -174,9 +162,15 @@ Before we start implementing the `Module` trait, there are several preparatory s
 
     ```
 
-3.  Define Config. In this case, config will contain admin and initial tokens:
+    As you can see, we derive the `borsh` serialization format for these messages. Unlike most serialization libraries,
+    `borsh` guarantees that all messages have a single "canonical" serialization, which makes it easy to reliably
+    hash and compare serialized messages.
+
+3.  Create a Config struct for the genesis configuration. In this case, the admin address and initial token distribution
+    are configurable
```
Comment (cemozerr) [inline examples/demo-nft-module/README.md:170]:
```suggestion
    are configurable:
```

File: examples/demo-nft-module/README.md [hunk 4]
```diff
@@ -485,5 +481,5 @@ Here's an example of how to do it with `AppTemplate` from `sov-default-stf`:
 }
 ```
 
-`AppTemplate` uses runtime to dispatch call during execution of `apply_batch` method.
-More details on how to setup rollup is available in [demo-rollup documentation](../demo-rollup/README.md)
+The `AppTemplate` uses `runtime` to dispatch calls during execution of the `apply_batch` method.
+Detailed instructions on how to set up a rollup can be found in the [`demo-rollup` documentation](../demo-rollup/README.md)
```
Comment (cemozerr) [inline examples/demo-nft-module/README.md:485]:
```suggestion
Detailed instructions on how to set up a rollup can be found in the [`demo-rollup` documentation](../demo-rollup/README.md).
```

File: examples/demo-stf/README.md [hunk 5]
```diff
@@ -19,21 +19,21 @@ with the rollup generically - so we can keep our modifications to the node as mi
 
 ## Implementing State Transition _Function_
 
-As you recall, the module system is primarily designed to help you implement the [State Transition Function
+As you recall, the Module System is primarily designed to help you implement the [State Transition Function
 interface](../../rollup-interface/specs/interfaces/stf.md).
 
 That interface is quite high-level - the only notion
-that it surfaces it that of a `blob` of rollup data. In the module system, we work at a much lower level - with
+that it surfaces it that of a `blob` of rollup data. In the Module System, we work at a much lower level - with
```
Comment (cemozerr) [inline examples/demo-stf/README.md:26]:
```suggestion
that it surfaces is that of a `blob` of rollup data. In the Module System, we work at a much lower level - with
```

File: examples/demo-stf/README.md [hunk 6]
```diff
@@ -19,21 +19,21 @@ with the rollup generically - so we can keep our modifications to the node as mi
...
 transactions signed by particular private keys. To bridge the gap, there's a system called an `AppTemplate`, which
 translates between the two layers of abstraction.
```
Comment (cemozerr) [inline examples/demo-stf/README.md:28]:
```suggestion
bridges between the two layers of abstraction.
```

File: module-system/README.md [hunk 7]
```diff
@@ -210,130 +211,8 @@ access the `Address` field from `Spec` - meaning that your bank logic doesn't ch
 Similarly, since each of the banks helper functions is automatically generic over a context, it's easy to define logic which
 can abstract away the distinctions between `zk` and `native` execution. For example, when a rollup is running in native mode
 its `Storage` type will almost certainly be [`ProverStorage`](./sov-state/src/prover_storage.rs), which holds its data in a
-merkle tree backed by RocksDB. But if you're running in zk mode the `Storage` type will instead be `ZkStorage`, which reads
+merkle tree backed by RocksDB. But if you're running in zk mode the `Storage` type will instead be [`ZkStorage`](./sov-state/src/zk_storage.rs), which reads
```
Comment (cemozerr) [inline module-system/README.md:214]:
```suggestion
Merkle tree backed by RocksDB. But if you're running in zk mode the `Storage` type will instead be [`ZkStorage`](./sov-state/src/zk_storage.rs), which reads
```

File: module-system/RPC_WALKTHROUGH.md [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+In the Module System, we provide handy macros to make it easy to generate RPC server implementations. In this document,
+we'll walk you through all of the steps that you need to take to enable RPC if you're implementing your rollup
+from scratch.
+
+There are 5 steps that need to be completed to enable RPC on the full node:
+
+1. Annotate you modules with `rpc_gen` and `rpc_method`.
+2. Annotate your `native` [`StateTransitionRunner`](../rollup-interface/src/state_machine/stf.rs) with the `expose_rpc` macro.
+3. Implement the `RpcRunner` trait on your `StateTransitionRunner`
```
Comment (cemozerr) [inline module-system/RPC_WALKTHROUGH.md:11]:
```suggestion
3. Implement the `RpcRunner` trait on your `StateTransitionRunner`.
```

File: module-system/RPC_WALKTHROUGH.md [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+4. Import and call `get_rpc_methods` in your full node implementation.
+5. Configure and start your RPC server in your full node implementation
```
Comment (cemozerr) [inline module-system/RPC_WALKTHROUGH.md:13]:
```suggestion
5. Configure and start your RPC server in your full node implementation.
```

File: module-system/RPC_WALKTHROUGH.md [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+}
+```
+
+Note that`expose_rpc` takes a tuple as argument, and each element of the tuple is a module with a concrete Context.
+
+### Step 3: Implement RpcRunner
+
+Next, we implement the `RpcRunner` trait on our `StateTransitionRunner`. If `expose_rpc` dictates which module RPCs we want to
+enable, `RpcRunner` dictates what kind of state they have access to. In this example, we'll expose the current state of our
+rollup by giving the RpcRunner a handle to our working database. However, we could just as easily use a different storage instance.
```
Comment (cemozerr) [inline module-system/RPC_WALKTHROUGH.md:78]:
```suggestion
rollup by giving the `RpcRunner` a handle to our working database. However, we could just as easily use a different storage instance.
```

**Pull request merged**